### PR TITLE
Test vectors: use tarballs, allow multiple spec versions for progressive spec updates

### PR DESCRIPTION
**Pull request by mratsim**
This updates the test vectors.

Now we use the EF tarballs to ensure that no git clone ever uses the EF quota.
We can also have different versions of the test vectors at the same time to allow progressive updates to v0.9.0 as requested by @tersec.

The only usage left of LFS is for the SSZ test vectors that are still using v0.8.1 format. It is against our own LFS quota however.

### CODE REVIEW

**Comment by stefantalpalaru** [comment]
Don't merge this until I'm done reviewing it.

File: scripts/setup_official_tests.sh [hunk 1]
```diff
@@ -33,8 +33,14 @@ COMPRESS_XZ="false"
 which 7z &>/dev/null && { DECOMPRESS_XZ="7z e -txz -bd -so"; COMPRESS_XZ="7z a -txz -an -bd -si -so"; }
 which xz &>/dev/null && { DECOMPRESS_XZ="xz -d -c -T 0"; COMPRESS_XZ="xz -c -T 0"; }
 
-download_lfs_files() {
-	[[ -z "$1" ]] && { echo "usage: download_lfs_files() subdir_name"; exit 1; }
+# download/unpack Ethereum Foundation tests functions
+source "$SUBREPO_DIR/scripts/download_functions.sh"
```
Comment (stefantalpalaru) [inline scripts/setup_official_tests.sh]: Why would you need this?

**Comment by tersec** [comment]
Azure CIs are showing:

```
--_curl_--https://github.com/ethereum/eth2.0-spec-tests/releases/download/v0.8.3/minimal.tar.gz
--_curl_--https://github.com/ethereum/eth2.0-spec-tests/releases/download/v0.8.3/mainnet.tar.gz

/d/a/1/s/jsonTestsCache /d/a/1/s
/d/a/1/s
ln: failed to create symbolic link 'tests/official/fixtures/tarballs': No such file or directory

##[error]Bash exited with code '1'.
```

File: scripts/setup_official_tests.sh (hunk 1 above)
Comment (mratsim) [reply scripts/setup_official_tests.sh]: That's for separating the download functions from their usage.

I need per-version flexibility for the CI compared to if someone does a manual clone and uses https://github.com/status-im/nim-eth2-official-tests/blob/master/download_test_vectors.sh

**Comment by mratsim** [comment]
It was not. bash is playing tricks on me:

DeepinScreenshot_select-area_20191031165627

**Pull request merged**